tests/unit/test_event.py

```diff
@@ -0,0 +1,158 @@
+import pytest
+from typing import List, Any
+from cdf_fabric_replicator.event import EventsReplicator
+from cognite.extractorutils.base import CancellationToken
+from cognite.extractorutils.statestore import LocalStateStore
+from cognite.client.data_classes import EventList, Event
+
+EARLY_CREATED_TIME = 1710503304020
+LATE_CREATED_TIME = 1710506651232
+
+@pytest.fixture(scope="session")
+def event_replicator(config):
+    stop_event = CancellationToken()
+    event_replicator = EventsReplicator(metrics=None, stop_event=stop_event)
+    event_replicator.config = config
+    event_replicator.client = event_replicator.config.cognite.get_cognite_client("test_event_replicator")
+    event_replicator.state_store = LocalStateStore(
+        event_replicator.config.extractor.state_store.local.path
+    )
+
+    return event_replicator
+
+
+@pytest.fixture
+def event_data():
+    return [
+        {
+            'external_id': 'Test_Early_Notification_1710502959241', 
+            'data_set_id': 3351080165451477, 
+            'start_time': 1710502959241, 
+            'end_time': 1710503079241, 
+            'type': 'Notification', 
+            'subtype': 'HE3', 
+            'description': 'Failed Instrumentation Fire extinguisher test', 
+            'metadata': {'description': 'Failed Instrumentation Fire extinguisher test', 'notification': '22081904', 'work_order_number': '22082901', 'notification_type': 'HE3', 'work_center': 'SUPV'}, 
+            'asset_ids': [5534771458240925], 
+            'id': 3631277804748961, 
+            'last_updated_time': 1710505316426, 
+            'created_time': EARLY_CREATED_TIME
+        },
+        {
+            'external_id': 'Test_Late_Notification_1710506390376', 
+            'data_set_id': 3351080165451477, 
+            'start_time': 1710506390376, 
+            'end_time': 1710506510376, 
+            'type': 'Notification', 
+            'subtype': 'HE3', 
+            'description': 'Failed Instrumentation Fire extinguisher test', 
+            'metadata': {'description': 'Failed Instrumentation Fire extinguisher test', 'notification': '22081904', 'notification_type': 'HE3', 'work_center': 'SUPV', 'work_order_number': '22082901'}, 
+            'asset_ids': [5534771458240925], 
+            'id': 5875589750425323, 
+            'last_updated_time': 1710506651232, 
+            'created_time': LATE_CREATED_TIME
+        }
+    ]
+
+@pytest.fixture
+def mock_get_events(mocker, event_data):
+    return_value = iter([EventList([Event(**data) for data in event_data])])
```

Comment:
A bit hard to follow what the logic is doing here with everything inline. Might be easier to follow if you break it out into a few intermediate steps.

Reply:
Modified.